example/newrepo/main.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// The newrepo command utilizes go-github as a cli tool for
+// creating new repositories.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+func main() {
+	token := os.Getenv("GITHUB_AUTH_TOKEN")
+	name := strings.Join(os.Args[1:], "-")
```

Comment:
This seems odd... why would a person call this with space-separated strings and not just hyphenate the single string themselves if that is the repo name they wanted?

Reply:
I agree, ideally users would only call this with a single argument but joining arguments felt like the most elegant way of handling cases where:
- There are no command line arguments (will panic if not handled),
- There are more than one command line arguments (subsequent arguments will be disregarded),

Do you recommend keeping it simple with `name := os.Args[1:][0]` and removing `if name == "" {
		log.Fatal("No name: New repos must be given a name")
	}
`? I could of course check `len(os.Args) > 1` but would have to do so before accessing `os.Args[1:][0]` so joining felt neater, albeit imperfect.

Let me know what you think. I really appreciate your input as I am still very new to the environment.

Reply:
If it were me, I would make it a flag.

Something like:

```go
var repoName = flag.String("repo", "", "Name of repo to create in authenticated user's GitHub account.")

func main() {
  flag.Parse()
  token := os.Getenv("GITHUB_AUTH_TOKEN")
  if token == "" {...}
  if *repoName == "" {
...
```

and then get rid of `name` and `os.Args` entirely.